core/src/main/scala/org/apache/spark/metrics/sink/StatsdSink.scala

```diff
@@ -46,23 +65,51 @@ private[spark] class StatsdSink(
     securityMgr: SecurityManager)
   extends Sink with Logging {
   import StatsdSink._
+  import collection.JavaConverters._
+
+  val reporterFactory = new StatsdReporterFactory()
 
-  val host = property.getProperty(STATSD_KEY_HOST, STATSD_DEFAULT_HOST)
-  val port = property.getProperty(STATSD_KEY_PORT, STATSD_DEFAULT_PORT).toInt
+  reporterFactory.host = property.getProperty(STATSD_KEY_HOST, STATSD_DEFAULT_HOST)
+  reporterFactory.port = property.getProperty(STATSD_KEY_PORT, STATSD_DEFAULT_PORT).toInt
+  reporterFactory.prefix = property.getProperty(STATSD_KEY_PREFIX, STATSD_DEFAULT_PREFIX)
 
   val pollPeriod = property.getProperty(STATSD_KEY_PERIOD, STATSD_DEFAULT_PERIOD).toInt
   val pollUnit =
     TimeUnit.valueOf(property.getProperty(STATSD_KEY_UNIT, STATSD_DEFAULT_UNIT).toUpperCase)
 
-  val prefix = property.getProperty(STATSD_KEY_PREFIX, STATSD_DEFAULT_PREFIX)
+  val excludes = property.getProperty(STATSD_KEY_EXCLUDES, STATSD_DEFAULT_EXCLUDES)
+  if(!excludes.isEmpty()) {
+    reporterFactory.setExcludes(ImmutableSet.copyOf(excludes.split(",")))
+  }
+
+  val includes = property.getProperty(STATSD_KEY_INCLUDES, STATSD_DEFAULT_INCLUDES)
+  if(!includes.isEmpty()) {
+    reporterFactory.setIncludes(ImmutableSet.copyOf(includes.split(",")))
+  }
+
+  val excludesAttributes = property.getProperty(STATSD_KEY_EXCLUDES_ATTRIBUTES, STATSD_DEFAULT_EXCLUDES_ATTRIBUTES).toUpperCase
+  if(!excludesAttributes.isEmpty()) {
+    reporterFactory.setExcludesAttributes(EnumSet.copyOf(excludesAttributes.split(",").map(attr => MetricAttribute.valueOf(attr)).toList.asJava))
+  }
+
+  val includesAttributes = property.getProperty(STATSD_KEY_INCLUDES_ATTRIBUTES, STATSD_DEFAULT_INCLUDES_ATTRIBUTES).toUpperCase
+  if(!includesAttributes.isEmpty()) {
+    reporterFactory.setIncludesAttributes(EnumSet.copyOf(includesAttributes.split(",").map(attr => MetricAttribute.valueOf(attr)).toList.asJava))
+  }
+
+  val useRegexFilter = property.getProperty(STATSD_KEY_USE_REGEX_FILTERS, STATSD_DEFAULT_USE_REGEX_FILTERS).toBoolean
+  val useSubstringMatching = property.getProperty(STATSD_KEY_USE_SUBSTRING_MATCHING, STATSD_DEFAULT_USE_SUBSTRING_MATCHING).toBoolean
 
   MetricsSystem.checkMinimalPollingPeriod(pollUnit, pollPeriod)
 
-  val reporter = new StatsdReporter(registry, host, port, prefix)
+  reporterFactory.setUseRegexFilters(useRegexFilter)
+  reporterFactory.setUseSubstringMatching(useSubstringMatching)
+
+  val reporter = reporterFactory.build(registry)
 
   override def start(): Unit = {
     reporter.start(pollPeriod, pollUnit)
-    logInfo(s"StatsdSink started with prefix: '$prefix'")
+    logInfo(s"StatsdSink started with prefix: '$reporterFactory.prefix'")
```

Comment:
this will most probably break all metric names and not quite important for metrics consumers. Let's just leave the original prefix which arrives here from configuration.

Reply:
Changed it to original prefix.